### ISSUE TITLE
Jackd server boot problem in Ubuntu 16.04

### ISSUE THREAD

**Issue by choforito84** [supercollider/supercollider#3270]
I could not create a server in supercollider due to an error with jackd. jackd couldn't create a server due to problems with the soundcard and pulseaudio. Running `pulseaudio --kill` in a terminal was useless.

How ever, the problem was solved by installing the package `pulseaudio-module-jack`. It is contained in Ubuntu package manager. I think you should add this package to the list of [packages needed to build from source](https://github.com/supercollider/supercollider/wiki/Installing-SuperCollider-from-source-on-Ubuntu).

Regards.

**Comment by snappizz** [comment]
`pulseaudio --kill` probably wasn't working because pulseaudio auto-restarts. if you add `autospawn = no` to `~/.config/pulse/client.conf` then the behavior is disabled.

the list of packages required for build is not really the appropriate place to add `pulseaudio-module-jack`. however, i agree that we should write documentation for linux users who want pulseaudio and jack working together.

**Comment by patrickdupuis** [comment]
This is strange to me. I don't remember having to install that package when I ran SC on Ubuntu or Debian. That package isn't installed on my current Fedora system either. Jack has no issue pushing pulseaudio aside and taking control of the soundcard. Maybe installing this package fixes the issue, but the real cause must be something else.

Are you using JACK1 or JACK2? I've always used JACK2. Maybe that's it?

**Comment by choforito84** [comment]
@patrickdupuis I'm using JACK2. I also had to change the soundcard used by JACK using qjackctl due to problems with the "default" configuration.

@snappizz You are right regarding the restart of pulseaudio. I could see it when listing all the processes of the computer. The online information regarding pulseaudio and jack working together is very little and not clear at all, documentation would be super helpful.

**Comment by brianlheim** [comment]
Changed the title of this issue to reflect that – is my description accurate?

**Comment by patrickdupuis** [comment]
@choforito84 Could you try removing the `pulseaudio-module-jack` package from your system and see if you can start JACK from QJackCtrl? I wonder if you simply need to configure JACK a little before trying to start it by booting the SC server.

 I've added a link below to a guide which explains how pulseaudio and JACK work together. The package you installed allows sound to be routed from pulseaudio to jack. You shouldn't have to install it in order to boot JACK on your system, nor should you need to kill pulseaudio in order to start JACK. Booting JACK should push pulseaudio aside, and allow it to come back when JACK stops.
 
https://help.ubuntu.com/community/UbuntuStudioPreparation#Pulse_Audio

**Comment by jamshark70** [comment]
I don't think this should go into a help file. It's not a SuperCollider problem. It's a system configuration problem. But, we could be nice to users and provide a wiki page that distills recommendations. choforito84's comment "The online information regarding pulseaudio and jack working together is very little and not clear at all" is a real and legitimate problem -- in Linux-land, it is indeed hard to find a straight answer.

So my vote is for a wiki page, and a help file could link to that.

(FWIW I've always had problems with dbus, pulse and jack. I ended up configuring pulse not to auto start. Instead, I have a startup script in qjackctl that launches and connects pulse to jack, and a shutdown script that kills pulse. So, on my system, if jack is not running, I have no audio from anywhere. When jack is running, all audio goes through it. But this, too, goes into the category of "opinions about how to do things in Linux, which might be completely wrong" :rofl: )

**Comment by snappizz** [comment]
i have something similar to that, but pulseaudio dies whenever i close my laptop lid so i have to manually restart it. (jack also dies sometimes, but less often) still in search of the perfect solution

**Comment by patrickdupuis** [comment]
wow! I fell fortunate that I haven't been experiencing anything like what you describe. Maybe it comes down to hardware? I don't know. I do however have Dbus disable in the qjackctrl settings.

**Comment by patrickdupuis** [comment]
http://jackaudio.org/faq/pulseaudio_and_jack.html

Interesting. I didn't know about the `pasuspender -- jackd` configuration in qjackctl.

**Comment by jamshark70** [comment]
> I wonder if you simply need to configure JACK a little before trying to start it by booting the SC server.

OTOH, a little note about this *could* go into a help file -- that we recommend users start JACK before booting the SC server. We kind of assume new users in Linux are familiar with real-time audio configuration, but users who are not familiar with it will just click "Boot server" and not know that jack may choke if not properly configured. If they are not using any other jack apps, all they see is "the SC server didn't start, but all my other apps make sound, so it's a problem with SC."

If the doc suggests starting jack in qjackctl first, then, if the system is not configured, users see that the problem occurs outside of SC.

This should go into docs about server booting, including but not limited to the Getting Started series.

**Comment by miguel-negrao** [comment]
> If the doc suggests starting jack in qjackctl first

qjackctl ignores the current dbus stored jackd settings and starts jack with its own internal settings, for this reason I recommend not using it when using jack with dbus. In that case I recommend using 'cadence' which does pick up the settings from dbus.

**Comment by jamshark70** [comment]
Aha, if qjackctl is not fully compatible with dbus, that probably explains why I never got along with dbus.

That means there are probably two cases that should be covered:

1. I just want to boot the SC server with minimum fuss, and I don't care about interacting with other apps (including PulseAudio).

2. Or, I want to set things up for seamless integration, and do it right.

... which is getting to be more than a little note in a help file. For the second, in Ubuntu, you have to add the KXStudio repositories before you can install cadence.

So it's starting to become what the OP *didn't* want: a more substantial list of "you could do it this way, or you could do it that way, or you could do it a totally different way."

(I'm not saying it wouldn't be useful -- just pointing out some potential traps.)

**Comment by ana0** [comment]
Am on xubuntu 16.04 and was also having to kill pulseaudio every time I started jack, this module has made them peacefully coexist, thank you!  +1 to pointing to linux system config tips in the docs somewhere
